### PR TITLE
Run wine of the MinGW tests

### PR DESCRIPTION
**Pull request by sjoelund**
### Purpose

Run wine for the Windows tests

### CODE REVIEW

File: testsuite/rtest [hunk 1]
```diff
@@ -9,22 +9,32 @@ use File::Temp qw/ tempfile /;
 (undef,$testTempFile) = tempfile("rtest$$.XXXXX", TMPDIR => 1, SUFFIX => ".tmp", UNLINK => 1);
 $cwd = getcwd;
 if ($cwd =~ m/(.*)testsuite\/(.+)$/) {
+  $PLATFORM1 = $ENV{'PLATFORM1'};
+  $PLATFORM2 = $ENV{'PLATFORM2'};
+  $PLATFORM3 = $ENV{'PLATFORM3'};
```
Comment (lochel) [inline testsuite/rtest]: Why are you not using `-platform=mingw` to override `$PLATFORM2` and `$PLATFORM3`?

Comment (sjoelund) [reply testsuite/rtest]: There wasn't such an option :D

File: testsuite/partest/runtests.pl [hunk 2]
```diff
@@ -172,6 +172,7 @@ sub read_makefile {
   my $dir = shift;
   my $header = shift;
 
+  return if($fast == 1 and $dir =~ m"/tlm"); # Skip tlm if -fast is given.
   return if($fast == 1 and $dir =~ m"/OMSysIdent"); # Skip OMSysIdent if -fast is given.
```
Comment (lochel) [inline testsuite/partest/runtests.pl]: We should run OMSysIdent for -fast.

**Comment by sjoelund** [comment]
@robbr48 seems we get fewer crashes now, but they still occur even with the latest changes:

```
+ tlm1d.lua ... execution failed
==229==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000 (pc 0x7f329700158f ...
#3 0x560ae84c6738 in PluginImplementer::Init(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, double, double, double, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) /var/lib/jenkins/workspace/OMSimulator_PR-607/OMTLMSimulator/common/Plugin/PluginImplementer.cc:202
#4 0x560ae846c2bc in oms::SystemTLM::connectToSockets(oms::ComRef, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) /var/lib/jenkins/workspace/OMSimulator_PR-607/src/OMSimulatorLib/TLM/SystemTLM.cpp:347
#5 0x560ae84686bb in oms::SystemTLM::stepUntil(double, void (*)(char const*, double, oms_status_enu_t)) /var/lib/jenkins/workspace/OMSimulator_PR-607/src/OMSimulatorLib/TLM/SystemTLM.cpp:271
#6 0x560ae856aacd in oms::Model::simulate() /var/lib/jenkins/workspace/OMSimulator_PR-607/src/OMSimulatorLib/Model.cpp:493
```

**Comment by sjoelund** [comment]
Hmmm... Did I just imagine things? I thought there was a mutex added to these calls? Or was that added in OMSimulator.git and not OMTLMSimulator.git?

**Comment by sjoelund** [comment]
Could be possible to create a thread-safe (locking or using GNU extensions) gethostbyname. Maybe also by changing TLMManagerComm.cc to not call gethostbyname (just use AF_INET)?

**Comment by sjoelund** [comment]
`valgrind --tool=drd` confirms my suspicion:
```
==5028== Thread 7:
==5028== Conflicting load by thread 7 at 0x08dde6c8 size 1
...
==5028==    by 0x5BC281B: gethostbyname (getXXbyYY.c:105)
==5028==    by 0x28FACD: TLMClientComm::ConnectManager(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, int) (in /home/martin/ssd/OMSimulator/install/linux/bin/OMSimulator)
==5028==    by 0x5494F57: MonitoringPluginImplementer::Init(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, double, double, double, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) (in /home/martin/ssd/OMSimulator/install/linux/lib/libomtlmsimulator.so)
==5028==    by 0x54BB1E8: InitializeTLMConnection(omtlm_CompositeModel&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) (in /home/martin/ssd/OMSimulator/install/linux/lib/libomtlmsimulator.so)
==5028==    by 0x54C3F2B: startMonitor(double, double, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, omtlm_CompositeModel&) (in /home/martin/ssd/OMSimulator/install/linux/lib/libomtlmsimulator.so)
==5028== Address 0x8dde6c8 is at offset 104 from 0x8dde660. Allocation context:
...
==5028==    by 0x5BC281B: gethostbyname (getXXbyYY.c:105)
==5028==    by 0x54B74FB: TLMManagerComm::CreateServerSocket() (in /home/martin/ssd/OMSimulator/install/linux/lib/libomtlmsimulator.so)
==5028==    by 0x54B206E: ManagerCommHandler::MonitorThreadRun() (in /home/martin/ssd/OMSimulator/install/linux/lib/libomtlmsimulator.so)
==5028==    by 0x54B71CD: ManagerCommHandler::thread_MonitorThreadRun(void*) (in /home/martin/ssd/OMSimulator/install/linux/lib/libomtlmsimulator.so)
==5028==    by 0x4C36413: ??? (in /usr/lib/valgrind/vgpreload_drd-amd64-linux.so)
...
```

**Comment by robbr48** [comment]
@sjoelund Mutex lock was implemented in OMSimulator. Actually I removed the mutex lock and made the initialization single-threaded instead.

**Comment by sjoelund** [comment]
Except it's not single threaded. The manager is started in a thread before the others...

**Comment by robbr48** [comment]
That's true. Only the submodel initialization is single-threaded. It could be problematic to add a mutex in OMTLMSimulator though, since it is actually designed for running submodels in separate processes. Running all submodels in the same process as the manager, as we do when simulating FMUs in OMSimulator, is in fact a special case.

**Pull request closed without merging**